google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java

```diff
@@ -0,0 +1,178 @@
+import com.google.cloud.securitycenter.v1beta1.SecurityCenterClient.ListAssetsPagedResponse;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import org.threeten.bp.Duration;
+import org.threeten.bp.Instant;
+import org.threeten.bp.LocalDateTime;
+import org.threeten.bp.ZoneId;
+import org.threeten.bp.temporal.ChronoUnit;
+
+/**
+ * Snippets for how to work with Assets in Cloud Security Command Center.
+ */
+public class AssetSnippets {
+
+  private final SecurityCenterClient securityCenterClient;
+  private final OrganizationName organizationName;
+
+  /**
+   * Filter that returns all projects in the organization
+   */
+  // [START  asset_resource_project_filter]
+  public static final String PROJECT_ASSET_FILTERS =
+      "security_center_properties.resource_type=\"google.cloud.resourcemanager.Project\"";
+  // [END  asset_resource_project_filter]
+
+
+  /**
+   * Create a new AssetSnippets object.
+   *
+   * @param client The client to use for contacting the service.
+   * @param organizationId The organization ID (this should be a numeric value, not the display name
+   * of the organization).
+   */
+  public AssetSnippets(SecurityCenterClient client, String organizationId) {
+    this.securityCenterClient = client;
+    // [START name_from_id]
+    this.organizationName = OrganizationName.of(organizationId);
+    // [END name_from_id]
+  }
+
+  /**
+   * Lists assets for an organization given meeting <code>filter</code> as of a specific instant in
+   * time.
+   *
+   * @param filter The filter that assets must meet (e.g. {@link #PROJECT_ASSET_FILTERS}).  If null,
+   * all assets in the organization are returned.
+   * @param asOf The instant in time to query for.  If null, current time is assumed
+   */
+
+  // [TARGET list_assets
+  // [VARIABLE "filter"]
+  // [VARIABLE "as_of"]
+  // [START list_assets ]
+  public ImmutableList<ListAssetsResult> listAssets(String filter,
+      Instant asOf) {
+    ListAssetsRequest.Builder request =
+        ListAssetsRequest.newBuilder().setParent(organizationName.toString());
+    // 
+    if (filter != null) {
+      request.setFilter(filter);
+    }
+    // Limits assets returned to a particular point in time.
+    if (asOf != null) {
+      request.getReadTimeBuilder().setSeconds(asOf.getEpochSecond()).setNanos(asOf.getNano());
+    }
+    ListAssetsPagedResponse response = securityCenterClient.listAssets(request.build());
+
+    // This creates one list for all assets.  If your organization has a large number of assets
+    // this can cause out of memory issues.  You can process them batches by returning
+    // the Iterable returned response.iterateAll() directly.
+    return ImmutableList.copyOf(response.iterateAll());
+  }
+  // [END list_assets ]
+
+  /**
+   * Run and print results from common queries.
+   */
+  void demoListAssets() {
+    // [ START demo_list_assets ] ]
```

Comment:
clean up whitespace and ]

Reply:
(there are a few of the same cleanups required in this file)

Reply:
I think I got them all, thanks for the thorough review.